### DISCUSSION TITLE
`Http::beforeSending` not working

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#50343] [Q&A] [answered]
![image](https://github.com/laravel/framework/assets/53290883/0ab7b17f-a9bd-4f44-b726-6bd7a4d2c135)

Shouldn't this work in this way? The `dump` never triggered

**Comment by henzeb** [comment, marked as answer]
Next time paste text instead of a screenshot. It's cumbersome to help, now I have to type it all over and show you what you misunderstood.

````php
Http::beforeSending(fn()=>dump('here'))->get('https://httpstat.us/200')->body();
````

beforeSending returns a pending request. it's not a global feature.

on a sidenote: There are better ways to dump... https://laravel.com/docs/10.x/http-client#dumping-requests
